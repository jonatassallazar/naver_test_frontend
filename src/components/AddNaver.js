import React, { useContext } from 'react';
import { useHistory } from 'react-router-dom';
import NaverForm from './NaverForm';
import store from '../store/store';
import { addNaver } from '../axios/instance';

const AddNaver = () => {
  let history = useHistory();

  const { dispatch } = useContext(store);

  return (
    <NaverForm
      onSubmit={(data) => {
        const result = addNaver(data);
        result
          .then(() => {
            history.push('/navers');
            dispatch({ type: 'ADD_NAVER' });
          })
          .catch((err) => {
            console.log(err);
          });
      }}
    />
  );
};

export default AddNaver;
