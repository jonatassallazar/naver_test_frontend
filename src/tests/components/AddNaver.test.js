import AddNaver from '../../components/AddNaver';
import 'jest-styled-components';
import customRender from './customRender';

test('should match LoginPage snapshot', () => {
  const { container } = customRender(<AddNaver />);

  expect(container).toMatchSnapshot();
});
