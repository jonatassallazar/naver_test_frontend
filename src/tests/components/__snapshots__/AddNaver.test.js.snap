// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match LoginPage snapshot 1`] = `
.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
}

.c6 {
  line-height: 2.4rem;
  height: 4rem;
  font-weight: 400;
  font-size: 1.6rem;
  font-family: Montserrat;
}

.c6::-webkit-input-placeholder {
  color: #9E9E9E;
}

.c6::-moz-placeholder {
  color: #9E9E9E;
}

.c6:-ms-input-placeholder {
  color: #9E9E9E;
}

.c6::placeholder {
  color: #9E9E9E;
}

.c5 {
  font-weight: 600;
  font-size: 1.4rem;
  line-height: 1.8rem;
  padding-bottom: 0.4rem;
}

.c2 {
  width: 1.8rem;
  height: 1.8rem;
}

.c8 {
  background: #212121;
  color: #ffffff;
  font-style: normal;
  font-weight: 600;
  font-size: 1.4rem;
  line-height: 2.4rem;
  text-align: center;
  border: none;
  padding: 1rem 5rem;
  width: 100%;
  font-family: inherit;
  cursor: pointer;
}

.c0 {
  margin: 0 auto;
  width: 50%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 3.2rem 0;
}

.c1 svg {
  width: 1.2rem;
  margin-right: 2.4rem;
}

.c1 h2 {
  font-weight: 600;
  font-size: 2.4rem;
  line-height: 3.6rem;
  margin: 1.4rem 0;
}

.c3 {
  width: 48%;
  margin-bottom: 2.4rem;
}

.c7 {
  position: absolute;
  right: 0;
  bottom: -5rem;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    width: 100%;
    padding: 1.4rem 2.4rem;
  }
}

@media (max-width:768px) {
  .c1 {
    margin: 0;
  }
}

@media (max-width:768px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c7 {
    padding-right: 2.4rem;
  }
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        href="/navers"
      >
        <svg
          class=" c2"
          version="1.1"
          viewBox="0 0 13 20"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M12.51 1.86998L10.73 0.0999756L0.839996 9.99998L10.74 19.9L12.51 18.13L4.38 9.99998L12.51 1.86998Z"
            fill="black"
          />
        </svg>
      </a>
      <h2>
        Adicionar Naver
      </h2>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <label
          class="c5"
        >
          Nome
        </label>
        <input
          class="c6"
          placeholder="Nome"
          value=""
        />
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <label
          class="c5"
        >
          Cargo
        </label>
        <input
          class="c6"
          placeholder="Cargo"
          value=""
        />
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <label
          class="c5"
        >
          Idade
        </label>
        <input
          class="c6"
          placeholder="Idade"
          value=""
        />
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <label
          class="c5"
        >
          Tempo de Empresa
        </label>
        <input
          class="c6"
          placeholder="Tempo de Empresa"
          value=""
        />
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <label
          class="c5"
        >
          Projetos que participou
        </label>
        <input
          class="c6"
          placeholder="Projetos que participou"
          value=""
        />
      </div>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <label
          class="c5"
        >
          URL da foto do Naver
        </label>
        <input
          class="c6"
          placeholder="URL da foto do Naver"
          value=""
        />
      </div>
    </div>
    <div
      class="c7"
    >
      <button
        class="c8"
        type="submit"
      >
        Salvar
      </button>
    </div>
  </form>
</div>
`;
